### PR TITLE
Add support for proxy_cache_path loader directives

### PR DESCRIPTION
**Pull request by carroarmato0**
Add support for loader_files, loader_sleep and loader_threshold for proxy_cache_path.

http://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_cache_path

```
Syntax:	proxy_cache_path path [levels=levels] [use_temp_path=on|off] keys_zone=name:size [inactive=time] [max_size=size] [manager_files=number] [manager_sleep=time] [manager_threshold=time] [loader_files=number] [loader_sleep=time] [loader_threshold=time] [purger=on|off] [purger_files=number] [purger_sleep=time] [purger_threshold=time];
Default:	—
Context:	http
```

Signed-off-by: Christophe Vanlancker <christophe.vanlancker@inuits.eu>

### CODE REVIEW

**Comment by wyardley** [comment]
Can you add tests to this?

**Comment by carroarmato0** [comment]
@wyardley managed to get the tests I wrote green.

File: templates/conf.d/nginx.conf.erb [hunk 1]
```diff
@@ -157,10 +157,10 @@ http {
 <% end -%>
 <% if @proxy_cache_path.is_a?(Hash) -%>
 <% @proxy_cache_path.sort_by{|k,v| k}.each do |key,value| -%>
-  proxy_cache_path        <%= key %> keys_zone=<%= value %> levels=<%= @proxy_cache_levels %> max_size=<%= @proxy_cache_max_size %> inactive=<%= @proxy_cache_inactive %><% if @proxy_use_temp_path %> use_temp_path=<%= @proxy_use_temp_path %><% end -%>;
+  proxy_cache_path        <%= key %> keys_zone=<%= value %> levels=<%= @proxy_cache_levels %> max_size=<%= @proxy_cache_max_size %> inactive=<%= @proxy_cache_inactive %><% if @proxy_use_temp_path %> use_temp_path=<%= @proxy_use_temp_path %><% end -%><% if @proxy_cache_loader_files %> loader_files=<%= @proxy_cache_loader_files %><% end -%><% if @proxy_cache_loader_sleep %> loader_sleep=<%= @proxy_cache_loader_sleep %><% end -%><% if @proxy_cache_loader_threshold %> loader_threshold=<%= @proxy_cache_loader_threshold %><% end -%>;
```
Comment (roidelapluie) [inline templates/conf.d/nginx.conf.erb]: Maybe we can use -%>; and split this us multiple lines for readibility?

File: templates/conf.d/nginx.conf.erb [hunk 2]
```diff
@@ -157,10 +157,10 @@ http {
...
 <% end -%>
 <% elsif @proxy_cache_path -%>
-  proxy_cache_path        <%= @proxy_cache_path %> levels=<%= @proxy_cache_levels %> keys_zone=<%= @proxy_cache_keys_zone %> max_size=<%= @proxy_cache_max_size %> inactive=<%= @proxy_cache_inactive %><% if @proxy_use_temp_path %> use_temp_path=<%= @proxy_use_temp_path %><% end %>;
+  proxy_cache_path        <%= @proxy_cache_path %> levels=<%= @proxy_cache_levels %> keys_zone=<%= @proxy_cache_keys_zone %> max_size=<%= @proxy_cache_max_size %> inactive=<%= @proxy_cache_inactive %><% if @proxy_use_temp_path %> use_temp_path=<%= @proxy_use_temp_path %><% end %><% if @proxy_cache_loader_files %> loader_files=<%= @proxy_cache_loader_files %><% end -%><% if @proxy_cache_loader_sleep %> loader_sleep=<%= @proxy_cache_loader_sleep %><% end -%><% if @proxy_cache_loader_threshold %> loader_threshold=<%= @proxy_cache_loader_threshold %><% end -%>;
```
Comment (roidelapluie) [inline templates/conf.d/nginx.conf.erb]: same

Comment (carroarmato0) [inline templates/conf.d/nginx.conf.erb]: Nginx unfortunately doesn't allow you do work with multi-lines (except for a few directives, but the proxy_cache_path is not one of them).

So I'm afraid I can't change that.

File: templates/conf.d/nginx.conf.erb (hunk 1 above)
Comment (carroarmato0) [inline templates/conf.d/nginx.conf.erb]: Same as below

Comment (roidelapluie) [inline templates/conf.d/nginx.conf.erb]: That is why I suggested -%>;

```
$ cat a.erb 
a <% if true %>b <% end -%>
c
$ erb -T - < a.erb 
a b c
```

**Pull request merged**